src/Raven.Server/Rachis/Leader.cs

```diff
@@ -777,42 +771,43 @@ private void EmptyQueue()
                                     }
                                     else
                                     {
-                                        if (result != null)
+                                        if (result != null && cmd.CtxResultWriter != null)
                                         {
-                                            result = GetConvertResult(cmd.Command)?.Apply(result) ?? cmd.Command.FromRemote(result);
+                                            cmd.CtxResultWriter.CopyResult(result);
+                                            result = null;
```

Comment:
I think we can remove this

Reply:
Do you mean to remove the setting to `null`?
I don't want to pass the blittable to the `TaskCompletionSource`

Reply:
okay, make sense, please add a comment about this

Reply:
done